apps/dot_ext/templates/oauth2_provider/application_detail.html

```diff
@@ -1,66 +1,162 @@
-  {% else %}
-      <p><pre>This value is not used when using a public client.</pre></p>
-  {% endif%}
-</div>
-
-<div class="ds-u-margin-y--2">
-  <p><b>{% trans "Client Type" %}</b></p>
-  <p><pre>{{ application.client_type }}</pre></p>
-</div>
-
-<div class="ds-u-margin-y--2">
-  <p><b>{% trans "Grant Type" %}</b></p>
-  <p><pre>{{ application.authorization_grant_type }}</pre></p>
-</div>
-
-<div class="ds-u-margin-y--2">
-  <p><b>{% trans "Redirect Uris" %}</b></p>
-  <pre>{{ application.redirect_uris }}</pre>
-</div>
-
-<div class="ds-u-margin-y--2">
-  <p><b>{% trans "Timestamps" %}</b></p>
-  <pre>Updated: {{ application.updated }} / Created: {{ application.created }}</pre>
-</div>
+<a class="cta-button ds-u-margin-y--1" href="{% url "oauth2_provider:delete" application.id %}" aria-label="Delete App" target="_self">
+    Delete Application
+    <i data-feather="trash-2"></i>
+  </a>
+{% endblock %}
 
+{% block content %}
+<!-- CI_TESTING:START -->
+<!-- test_dot_ext_application_detail -->
+<!-- CI_TESTING:END -->
 
-<a class="ds-c-button  ds-c-button--default" href="{% url "oauth2_provider:update" application.id %}">{% trans "Edit" %}</a>
-<a class="ds-c-button  ds-c-button--danger" href="{% url "oauth2_provider:delete" application.id %}">{% trans "Delete" %}</a>
+<!-- <div class="bb-width--75 ds-u-md-margin-left--3 ds-u-lg-margin-left--4 ds-u-xl-margin-left--5"></div> -->
 
-<div class="ds-u-margin-y--2">
-  <a class="btn" href="{% url "oauth2_provider:list" %}">{% trans "Go Back" %}</a>  
+<div class="container">
+    <div class="ds-l-row ds-u-justify-content--center ds-u-margin-y--4">
+      <div class="ds-l-lg-col--11 ds-l-md-col--11 ds-l-sm-col--12">
+          <div class="bb-width--75">
+              <!-- App Credential Section -->
+              <h2 class="ds-u-margin-bottom--2">App Credentials</h2>
+            
+              <div class="bb-c-card sandbox-app-card">
+                <!-- Application Credentials -->
+                <form class="app-credentials">
+                  <span class="copy-confirmation" id="confirm-id-{{ application.client_id }}" aria-live="assertive" aria-role="alert">ID Copied to Clipboard</span>
+                  <span class="copy-confirmation" id="confirm-secret-{{ application.client_id }}" aria-live="assertive" aria-role="alert">Secret Copied to Clipboard</span>
+                  <a class="app-credentials-toggle" id="{{ application.client_id }}" tabindex="0">Show/Hide Credentials</a>
+                  <label for"id-{{ application.client_id }}">Client ID</label>
+                  <input type="password" value="{{ application.client_id }}" id="id-{{ application.client_id }}" onclick="copyCredential(this.id)" />
+                  <label for"secret-{{ application.client_secret }}">Client Secret</label>
+                  <input type="password" value="{{ application.client_secret }}" id="secret-{{ application.client_id }}" onclick="copyCredential(this.id)" />
```

Comment:
Yep! So, basically since there are going to be multiple applications at any given time, it made it easier to toggle the `id` and `secret` for each individual app. I think it also made targeting the `id` and `secret` easier from a traversal standpoint. Like, because of the structure of the HTML, I can't just grab a sibling element, so being able to narrow it down to each individual's app id helped me, conceptually and practically. 

Not saying there isn't a better way to do it, but it made a lot of sense to me. Happy to work this from a different angle if you think that's worthwhile now.

Reply:
that makes total sense, and it became much more clear as I continued to read the code.